### PR TITLE
Use lazy ghosts unconditionally

### PR DESCRIPTION
**Pull request by greg0ire**
A nice consequence of this is that we can almost get rid of `doctrine/common` now. I think we might only depend on [`ClassUtils`](https://github.com/doctrine/common/blob/3.4.x/src/Util/ClassUtils.php) now.

### CODE REVIEW

File: lib/Doctrine/ORM/Proxy/ProxyFactory.php [hunk 1]
```diff
@@ -191,14 +164,13 @@ public function __construct(
     }
 
     /**
-     * {@inheritDoc}
+     * @param class-string $className
+     * @param array<mixed> $identifier
+     *
+     * @return InternalProxy
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Proxy/ProxyFactory.php:170]: Using a native return type here will involve modify some tests that mock calls to `getProxy` and make them return actual entities.

**Comment by greg0ire** [comment]
Cc @nicolas-grekas

File: lib/Doctrine/ORM/Configuration.php [hunk 2]
```diff
@@ -581,28 +577,19 @@ public function setSchemaIgnoreClasses(array $schemaIgnoreClasses): void
...
+     * @return true
+     */
     public function isLazyGhostObjectEnabled(): bool
     {
-        return $this->attributes['isLazyGhostObjectEnabled'] ?? false;
+        return true;
     }
 
+    /** To be deprecated in 3.1.0 */
     public function setLazyGhostObjectEnabled(bool $flag): void
```
Comment (derrabus) [inline lib/Doctrine/ORM/Configuration.php:592]: We should throw if `$flag === false`.

File: lib/Doctrine/ORM/Proxy/ProxyFactory.php (hunk 1 above)
Comment (derrabus) [reply lib/Doctrine/ORM/Proxy/ProxyFactory.php:170]: I think we should fix those tests, but that could be done in a follow-up.

**Review by derrabus — COMMENTED** [review]
Please modify composer.json and make `symfony/var-exporter` a mandatory dependency.

**Comment by greg0ire** [comment]
> I think we should fix those tests, but that could be done in a follow-up.

Yeah me too… but I haven't found a clear solution so far.

File: composer.json [hunk 3]
```diff
@@ -33,7 +33,8 @@
         "doctrine/lexer": "^2.1 || ^3",
         "doctrine/persistence": "^3.1.1",
         "psr/cache": "^1 || ^2 || ^3",
-        "symfony/console": "^5.4 || ^6.0 || ^7.0"
+        "symfony/console": "^5.4 || ^6.0 || ^7.0",
+        "symfony/var-exporter": "^5.4 || ^6.2"
```
Comment (derrabus) [inline composer.json]:
```suggestion
        "symfony/var-exporter": "^6.2"
```

Lazy ghost objects don't exist in v5.4. We need version 6.

Comment (greg0ire) [reply composer.json]: :thinking: this makes me notice that for some reason, we don't have a job with `--prefer-lowest` on this branch.

Comment (greg0ire) [reply composer.json]: @derrabus apparently you removed it in https://github.com/doctrine/orm/pull/9069 … do you recall why?

Comment (derrabus) [reply composer.json]: No, let's add it back.

Comment (greg0ire) [reply composer.json]: Done in #10971

**Review by derrabus — APPROVED** [review]

**Pull request merged**